diuf/sudoku/gui/GenerateDialog.java

```diff
@@ -527,6 +527,168 @@ public String getnotMaxTechnique2() {
+                return 0.0;
+            }
+            @Override
+            public double getnotMaxDifficulty3() {
+                return 0.0;
+            }
+			public String getexcludeTechnique1() {
+				return "Skyscraper";
+			}
+            @Override
+			public String getexcludeTechnique2() {
+				return "";
+			}
+            @Override
+			public String getexcludeTechnique3() {
+				return "";
+			}
+			public String getincludeTechnique1() {
+				return "";
+			}
+            @Override
+			public String getincludeTechnique2() {
+				return "";
+			}
+            @Override
+			public String getincludeTechnique3() {
+				return "";
+			}
+			public String getnotMaxTechnique1() {
+				return "";
+			}
+            @Override
+			public String getnotMaxTechnique2() {
+				return "";
+			}
+            @Override
+			public String getnotMaxTechnique3() {
+				return "";
+			}
+        },
+		ThreeStrongLinks {
+
+            @Override
+            public double getMinDifficulty() {
+                return 5.4;
+            }
+
+            @Override
+            public double getMaxDifficulty() {
+                return 5.6;
+            }
+            @Override
+            public double getincludeDifficulty1() {
+                return 0.0;
+            }
+            @Override
+            public double getincludeDifficulty2() {
+                return 0.0;
+            }
+            @Override
+            public double getincludeDifficulty3() {
+                return 0.0;
+            }
+            @Override
+            public double getexcludeDifficulty1() {
+                return 0.0;
+            }
+            @Override
+            public double getexcludeDifficulty2() {
+                return 0.0;
+            }
+            @Override
+            public double getexcludeDifficulty3() {
+                return 0.0;
+            }
+            @Override
+            public double getnotMaxDifficulty1() {
+                return 0.0;
+            }
+            @Override
+            public double getnotMaxDifficulty2() {
+                return 0.0;
+            }
+            @Override
+            public double getnotMaxDifficulty3() {
+                return 0.0;
+            }
+			public String getexcludeTechnique1() {
+				return "";
+			}
+            @Override
+			public String getexcludeTechnique2() {
+				return "";
+			}
+            @Override
+			public String getexcludeTechnique3() {
+				return "";
+			}
+			public String getincludeTechnique1() {
+				return " 10";
+			}
+            @Override
+			public String getincludeTechnique2() {
+				return "";
+			}
+            @Override
+			public String getincludeTechnique3() {
+				return "";
+			}
+			public String getnotMaxTechnique1() {
+				return "";
+			}
+            @Override
+			public String getnotMaxTechnique2() {
+				return "";
+			}
+            @Override
+			public String getnotMaxTechnique3() {
+				return "";
+			}

```

Comment:
no, it isn't messy ;)

Reply:
This is all my doing. I can't blame anybody else :) ... The idea is to have more control on what to generate 
Command line should be easier to control. GUI currently has Drop down menu with a general theme of generated puzzle. More can be done if there were drop menus / input field  for each sub category (rating, technique to include , …)

Reply:
My solution would be TABLE with one row per (sub-)technique and many columns - rating, rating rules, names, short/long names, html description template, java class and constuctor parameters, etc, etc. even including nested tables if needed.
Then you should program GUI just what column to display.

Reply:
I uploaded a generateDiff.ods file which I can use to create the JTABLE or TABLE when I have time